core/shared/src/main/scala/cats/parse/Rfc5234.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2020 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package cats.parse
+
+/** Parsers for the common rules of RFC5234. These rules are
+  * referenced by several RFCs.
+  *
+  * @see [[https://tools.ietf.org/html/rfc5234]]
+  */
+object Rfc5234 {
+
+  /** A-Z and a-z, without diacritics
+    */
+  val alpha: Parser1[Char] =
+    Parser.charIn('A' to 'Z').orElse1(Parser.charIn('a' to 'z'))
+
+  /** `0` or `1`
+    */
+  val bit: Parser1[Char] =
```

Comment:
Was tempted to make this return a `Boolean`.  Similar considerations for `digit` and `hexdigit`.  Maybe it's good to have these raw versions, and maps for the obvious interpretations.